api/v3/Activity/Getmonthswithactivities.php

```diff
@@ -0,0 +1,71 @@
+<?php
+use CRM_Civicase_ExtensionUtil as E;
+
+/**
+ * Activity.Getmonthswithactivities API specification
+ *
+ * @param array $spec description of fields supported by this API call
+ * 
+ * @return void
+ * @see http://wiki.civicrm.org/confluence/display/CRMDOC/API+Architecture+Standards
+ */
+function _civicrm_api3_activity_Getmonthswithactivities_spec(&$spec) {
+  $allowed = ['activity_date_time', 'activity_status_id', 'case_id'];
+  $all = civicrm_api3('Activity', 'getfields', array('api_action' => 'get'))['values'];
+  
+  $spec = array_filter($all, function ($name) use ($allowed) {
+    return in_array($name, $allowed);
+  }, ARRAY_FILTER_USE_KEY);
+}
+
+/**
+ * Returns list of unique [MM, YYYY] month-year pair with at least an activity
+ * 
+ * @method Activity.Getmonthswithactivities API
+ *
+ * @param array $params
+ * @return array API result with list of months
+ * @see civicrm_api3_create_success
+ * @throws API_Exception
+ */
+function civicrm_api3_activity_Getmonthswithactivities($params) {
+  try {
```

Comment:
Why do we need a `try/catch` block here? Cant we handle the errors manually.
For example, https://github.com/compucorp/uk.co.compucorp.civicase/blob/C51-269-getmonthswithactivities-endpoint/api/v3/Case/Getdetails.php#L233